src/core/stdcpp/exception.d

```diff
@@ -112,5 +112,13 @@ class bad_exception : exception
 {
 @nogc:
     ///
-    this(const(char)* message = "bad exception") { super(message); }
+    this(const(char)* message = "bad exception") nothrow { super(message); }
+
+    version (GenericBaseException)
+    {
+        ///
+        ~this() nothrow;
```

Comment:
Not declaring the dtor means it gets an implicit one (base class has one), and that one collides with the C++ one. [For MSVC, everything is defined inline in the C++ headers, so the implicit one does the job.]